tests/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixerTest.php

```diff
@@ -0,0 +1,270 @@
+                    "c.*" => "array",
+                    "d" => "int",
+                    "d.*" => "string",
+                ];',
+            '<?php [
+                    "c.*" => "array",
+                    "b" => "array",
+                    "d.*" => "string",
+                    "b" . "a" => "bla",
+                    "a" . "a" => "bla",
+                    "d" => "int",
+                    "a.*" => "int",
+                ];',
+            [
+                'special_keys_placement' => 'top',
+            ],
+        ];
+
+        yield 'old-style array with string keys' => [
+            '<?php array("a" => "a", "b" => "b", "c" => "c");',
+            '<?php array("c" => "c", "a" => "a", "b" => "b");',
+        ];
+
+        yield 'old-style array with integer keys' => [
+            '<?php array(1 => "one", 2 => "two", 3 => "three");',
+            '<?php array(3 => "three", 1 => "one", 2 => "two");',
+        ];
+
+        yield 'old-style array with mixed keys' => [
+            '<?php array("a" => "a", "b" => "b", 1 => "one", 2 => "two");',
+            '<?php array("b" => "b", 1 => "one", 2 => "two", "a" => "a");',
+        ];
+
+        yield 'array with string keys' => [
+            '<?php ["a" => "a", "b" => "b", "c" => "c"];',
+            '<?php ["c" => "c", "a" => "a", "b" => "b"];',
+        ];
+
+        yield 'array with mixed keys, starting with integer' => [
+            '<?php ["a" => "a", "b" => "b", "c" => "c", 0 => 0, 1 => 1, 2 => 2];',
+            '<?php [2 => 2, "c" => "c", 0 => 0, "a" => "a", "b" => "b", 1 => 1];',
+        ];
+
+        yield 'array with integer keys' => [
+            '<?php [1 => "one", 2 => "two", 3 => "three"];',
+            '<?php [3 => "three", 1 => "one", 2 => "two"];',
+        ];
+
+        yield 'array with mixed keys, starting with string' => [
+            '<?php ["a" => "a", "b" => "b", 1 => "one", 2 => "two"];',
+            '<?php ["b" => "b", 1 => "one", 2 => "two", "a" => "a"];',
+        ];
+
+        yield 'function calls in keys, including first' => [
+            '<?php ["a" => "value3", "b" => "value4", "c" => "value5", foo() . " baz" => "value1", bar() => "value2"];',
+            '<?php [foo() . " baz" => "value1", "b" => "value4", bar() => "value2", "a" => "value3", "c" => "value5"];',
+        ];
+
+        yield 'function calls in keys, but first key as simple string' => [
+            '<?php ["a" => "value3", "b" => "value4", kar() => "value4", foo() . " baz" => "value1", bar() => "value2"];',
+            '<?php ["b" => "value4", kar() => "value4", foo() . " baz" => "value1", "a" => "value3", bar() => "value2"];',
+        ];
+
+        yield 'static call as one of values' => [
+            '<?php array("a" => Arr::get($bla, "environment.bla"), "b" => "value3", "c" => "value1");',
+            '<?php array("c" => "value1", "a" => Arr::get($bla, "environment.bla"), "b" => "value3");',
+        ];
+
+        yield 'multiline old-style array' => [
```

Comment:
we need tests with single- and multi-line comments, to make sure comments are moved properly

Reply:
There is test case `comment between items` and it seems like array items are moved, but comments not, so after the fix comment is above different item... Do I understand correctly you would like to move comment along with the array item?

Reply:
Overall to handle different type of comments (single/multi-lined , different comment syntaxes) as well as their stickiness to items they comment on